src/app/state/state/GlobalStateProvider.tsx

```diff
@@ -18,6 +18,14 @@ const Provider: React.FC = ({ children }) => {
     }
   }
 
+  // TODO: Use globalstate-hooks Config.shouldIndex
+  const { index: casesIndex } = casesActions
+  const { index: caseTypesIndex } = caseTypesActions
+  useEffect(() => {
+    casesIndex()
+    caseTypesIndex()
```

Comment:
I don't think this is a good idea.

This will just fetch everything at startup, which I don't think is scalable in the future (when we'll have a lot more to fetch). 
I think components that need the data should be able to fetch it themselves.

Please hava a look at a Apollo / GraphQL example:
https://www.apollographql.com/docs/react/data/queries/#caching-query-results

The `useQuery()` will call the API is necessary. 

```
function DogPhoto({ breed }) {
  const { loading, error, data } = useQuery(GET_DOG_PHOTO, {
    variables: { breed },
  });

  if (loading) return null;
  if (error) return `Error! ${error}`;

  return (
    <img src={data.dog.displayImage} style={{ height: 100, width: 100 }} />
  );
}
```

Reply:
Yes, I agree. Eventually all `index` calls could become more lazy. For now this is more than performant and solves the problem of fetching the same resource more than once. Will keep lazy initialisation in mind when implementing pagination and polling.

Reply:
Awesome :-)